src/lsm/groove.zig

```diff
@@ -844,7 +917,10 @@ pub fn GrooveType(
         /// Insert the value into the objects tree and its fields into the index trees.
         fn insert(groove: *Groove, object: *const Object) void {
             groove.objects.put(object);
-            if (has_id) groove.ids.put(&IdTreeValue{ .id = object.id, .timestamp = object.timestamp });
+            if (has_id) {
+                groove.ids.put(&IdTreeValue{ .id = object.id, .timestamp = object.timestamp });
+                groove.ids.update_key_range(object.id);
```

Comment:
Yes, we only update it for the id tree! This is because we are only consuming the tree range during negative lookups (in `prefetch_enqueue_negative_lookup_optimized`), which is invoked only on grooves which have an `id` tree. Also, since we're updating the key range on every key insertion into the tree doing it for all trees would be wasteful.

PS: The logic to update the tree's shape will be moved to `prefetch_setup` when Fed's changes come in, this would amortize the cost of updating the shape!